### ISSUE TITLE
[Bug] Manifest parsing yields None macro dependency

### ISSUE THREAD

**Issue by mazinesy** [dbt-labs/dbt-core#9261]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Given an unknown macro under `models.columns.tests` :
```yml
models:
  - name: second_model
    description: "The second model"
    columns:
      - name: fun
        tests:
          - not_null
          - unknow_macro  # this is either invalid or a typo
```

The manifest file yields a `None` in the `node.depends_on.macros` for the test node.

- `dbt parse` succeeds even though the manifest is invalid.
- `dbt compile` fails as expected, but it fails during the `SQL` generation/querying when it tries to render the `None` macro.

### Expected Behavior

Both `parse` and `compile` commands fail because the `manifest` is invalid.

Currently, `compile` fails later on (SQL step), but it could (should ?) fail earlier.

### Steps To Reproduce

I have a closed [PR](https://github.com/dbt-labs/dbt-core/pull/9233) to document this.

tldr; Every `Macro#get_macro_id` is followed by an `if` statement but the one in `SchemaGenericTests`.

### Which database adapter are you using with dbt?

postgres

### Additional Context

I added tests in the `compile` fixtures because I did not know where it happened. Now that the debugging is done, I believe the actual tets should live somewhere alongside the `Manifest` ones.

**Comment by mazinesy** [comment]
:wave: @dbeatty10 I don't mind reopening the PR, I think the tests in there are kind of poor, and should live somewhere else. I also don't know if there are some macros that are dynamically resolve at runtime, but I doubt it is the case.

If you feel comfortable with the changes, I can reopen the PR, move the tests at the right place and we can go from there

**Comment by dbeatty10** [comment]
That sounds great @mazinesy 🚀

One of our engineers will assigned to review and help take a look at the edge cases, location of the tests, etc.
